HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,268 @@
+    if (!backToBackWithMet) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtThresh_) | (!withinEtaRange)) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // Cuts related to eta and phi width of HF showers:
+    // Sigma eta-phi based cut + central strip size cut
+    // We'll apply the sigma eta+phi dependent cut only if requested via the config file
+    if (applySigmaEtaPhiCut_) {
+      float sieie = v_sigmaEtaEta[ijet];
+      float sipip = v_sigmaPhiPhi[ijet];
+
+      // Sigma eta eta and sigma phi phi
+      bool passSigmaEtaPhiCut = sieie - sipip < sigmaEtaPhiDiffThresh_;
+
+      if (applyCornerCut_) {
+        bool inCorner = (v_sigmaEtaEta[ijet] < 0.02) & (v_sigmaPhiPhi[ijet] < 0.02);
+        passSigmaEtaPhiCut = passSigmaEtaPhiCut & (!inCorner);
+      }
+      if (!passSigmaEtaPhiCut) {
+        continue;
+      }
+    }
+
+    if (applyStripSizeCut_) {
+      // Cut related to central strip size
+      int cess = v_centralEtaStripSize[ijet];
+      bool passCentralStripSizeCut = cess < centralEtaStripSizeThresh_;
+      if (!passCentralStripSizeCut) {
+        continue;
+      }
+    }
+
+    cleaned_jets->push_back(jet);
+  }
+
+  iEvent.put(std::move(cleaned_jets));
+}
+
+// ------------ method called once each stream before processing any runs, lumis or events  ------------
+void HLTHFJetCleaner::beginStream(edm::StreamID) {
+  // please remove this method if not needed
+}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void HLTHFJetCleaner::endStream() {
+  // please remove this method if not needed
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+HLTHFJetCleaner::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+HLTHFJetCleaner::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+HLTHFJetCleaner::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+HLTHFJetCleaner::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void HLTHFJetCleaner::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
```

Comment:
This needs to be updated to include all the configuration parameters of this plugin, each with its default value. Otherwise, (among other things) ConfDB will never know them.

Reply:
Implemented in [2a4e694](https://github.com/cms-sw/cmssw/pull/37539/commits/2a4e6947671e4c8730dd5dd4c1813d74ee793c1c)